### ISSUE TITLE
Add 'name' attribute to all sub-classes of ActionMenuItem

### ISSUE THREAD

**Issue by robslotboom** [wagtail/wagtail#5154]
### Issue Summary

To use the hook “construct_page_action_menu” it would be nice to be able to identify the menu items in the list. So I suggest to ad an id to the ActionMenuItem.

For instance:
```
class PublishMenuItem(ActionMenuItem):
   id = 'publish'

class SubmitForModerationMenuItem(ActionMenuItem):
    id = 'submit_for_moderation'

etc.
```
Then you can do things like this:

```
@hooks.register('construct_page_action_menu')
def protect_homepage(menu_items, request, context):
	page = context.get('page')	
	if page.specific = HomePage: 	
		new_items = []
		for item in menu_items:
			if item.id == 'unpublish' or item.id == 'delete':
				pass
			else:
				new_items.append(item)
		menu_items[:] = [item for item in new_items]
```

### Technical details

* Wagtail version: 2.4

**Comment by lb-** [comment]
@robslotboom  - thank you for submitting this request.

Upon some investigation, it appears that `ActionMenuItem` already has the attribute `name` but it was only set on `SubmitForModerationMenuItem`.

The docs referring to the hook you referenced even used item.name to 'remove' something from the menu item list.

http://docs.wagtail.io/en/v2.4/reference/hooks.html#construct-page-action-menu

So to align with this use case it would be best we used name instead of id but just added it to all core `ActionMenuItems` that are in use.

I will submit a PR with this fix for future.

In the mean time, you can do some (not as clean) comparison of items by doing something like the following:

```
@hooks.register('construct_page_action_menu')
def protect_homepage(menu_items, request, context):
	page = context.get('page')	
	if page.specific = HomePage: 	
		new_items = []
		for item in menu_items:
                        # read the class name via .__class__.__name__
			if item.__class__.__name__ in ['DeleteMenuItem', 'UnpublishMenuItem']:
				pass
			else:
				new_items.append(item)
		menu_items[:] = [item for item in new_items]
```

Hope that helps.

**Comment by robslotboom** [comment]
Hi Ben,

Thank you for the reply.
I knew about the name attr and saw it wasn’t set, except for the Submit...
But for some reason I preferred an ID.

> In the mean time, you can do some (not as clean) comparison of items by doing something like the following:

Excellent, clean enough and without the need to change the core :-)

Regards from Doesburg,

Rob

**Comment by gasman** [comment]
Fixed in #5157
